### PR TITLE
Backports 6.0.5 20210113/v1

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

- https://redmine.openinfosecfoundation.org/issues/4896
- https://redmine.openinfosecfoundation.org/issues/4837
- https://redmine.openinfosecfoundation.org/issues/4822

- https://redmine.openinfosecfoundation.org/issues/4818
https://github.com/OISF/suricata/commit/0e70958e67398f599b77ba991dc631e025cae5f4 backported,
https://github.com/OISF/suricata/commit/c1bffa9545b8aa9d0fc64ac6511edd34919135d7 was already in the branch.

- https://redmine.openinfosecfoundation.org/issues/4744
- https://redmine.openinfosecfoundation.org/issues/4467
- https://redmine.openinfosecfoundation.org/issues/4933

Should still likely fail (only Fuzz (address)) because of missing fix for https://redmine.openinfosecfoundation.org/issues/4894

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | tlpr1_stats_chk |
| .flow.memuse | 512225408 | 544894528 | 94.0% |
| | generic_stats_chk |
| .capture.kernel_drops | 0 | 92961 | 0.0% |
| .tcp.reassembly_gap | 80952 | 97967 | 82.63% | 

Pipeline 5687

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -307,6 +307,158 @@ jobs:
...
+          ac_cv_func_realloc_0_nonnull: "yes"
+          ac_cv_func_malloc_0_nonnull: "yes"
+      - run: make -j2
+      - run: ASAN_OPTIONS="detect_leaks=0" ./src/suricata -u -l .
+      - name: Extracting suricata-verify
+        run: tar xf prep/suricata-verify.tar.gz
+      - name: Running suricata-verify
+        run: python3 ./suricata-verify/run.py
+
   fedora-33:
```
Comment (inashivb) [inline .github/workflows/builds.yml:462]: Remove 32 & 33 as they are EOL. Suggested by Jason on 5.0.x PR

File: src/app-layer-detect-proto.c [hunk 2]
```diff
@@ -1985,6 +1989,11 @@ int AppLayerProtoDetectConfProtoDetectionEnabled(const char *ipproto,
     SCReturnInt(enabled);
 }
 
+int AppLayerProtoDetectConfProtoDetectionEnabled(const char *ipproto, const char *alproto)
+{
+    return AppLayerProtoDetectConfProtoDetectionEnabledDefault(ipproto, alproto, true);
+}
+
```
Comment (inashivb) [inline src/app-layer-detect-proto.c:1996]: Drop commit

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | generic_stats_chk |
| .capture.kernel_drops | 0 | 92961 | 0.0% |
| .tcp.reassembly_gap | 80952 | 97967 | 82.63% | 

Pipeline 5700

**Comment by inashivb** [comment]
Replaced w #6806

**Pull request closed without merging**